addons/event/static/src/client_action/event_registration_summary_dialog.js

```diff
@@ -36,20 +57,27 @@ export class EventRegistrationSummaryDialog extends Component {
     }
 
     get needManualConfirmation() {
-        return this.registration.status === "need_manual_confirmation";
+        return this.registrationStatus.value === "need_manual_confirmation";
     }
 
     async onRegistrationConfirm() {
         await this.orm.call("event.registration", "action_set_done", [this.registration.id]);
         this.registrationStatus.value = "confirmed_registration";
     }
 
-    onRegistrationPrintPdf() {
-        this.actionService.doAction({
-            type: "ir.actions.report",
-            report_type: "qweb-pdf",
-            report_name: `event.event_registration_report_template_badge/${this.registration.id}`,
-        });
+    async onRegistrationPrintPdf() {
+        if (this.printSettings.iotPrinterId) {
+            const reportName = `event.event_report_template_esc_label_${this.registration.badge_format}_badge`;
+            const [{ id: reportId }] = await this.orm.searchRead("ir.actions.report", [["report_name", "=", reportName]], ["id"]);
+            const printJobArguments = [reportId, [this.registration.id], null, uuid()]
+            await this.env.services.iot_websocket.addJob([this.printSettings.iotPrinterId], printJobArguments)
```

Comment:
As discussed just as for the reports printing we need to go through the longpolling / http request on the local network first before trying the websocket connection as it will take more time. Odo suggestion: try to ping the IoT Box's ip with a very short timeout. If the ping succeeds, send it via the longpolling. Otherwise, send it via the websocket straigt away

Reply:
Now has a 1 second ping, using longpolling if succeeding